### PR TITLE
Singlecell improvements

### PR DESCRIPTION
**Pull request by airenzp**
## Description

Displayed plots two or three in a row, customized chart buttons, customized colors and legends

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [x] Tests: added and/or passed unit and integration tests, or N/A
- [x] Todos: commented or documented, or N/A
- [x] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

File: client/plots/singleCellPlot.js [hunk 1]
```diff
@@ -25,26 +25,30 @@ class singleCellPlot {
...
 		const controlsDiv = this.opts.holder.insert('div').style('display', 'inline-block')
-		this.mainDiv = this.opts.holder
-			.insert('div')
-			.style('display', 'inline-block')
-			.style('vertical-align', 'top')
-			.style('padding', '10px')
-		this.sampleDiv = this.mainDiv.insert('div').style('display', 'inline-block')
-
+		this.mainDiv = this.opts.holder.insert('div').style('display', 'inline-block').style('vertical-align', 'top')
+		if (appState.vocab.dslabel == 'GDC') {
```
Comment (xzhou82) [inline client/plots/singleCellPlot.js:29]: have to change it later

File: client/plots/singleCellPlot.js [hunk 2]
```diff
@@ -120,13 +120,13 @@ class singleCellPlot {
...
+		this.dom.tableDiv.selectAll('*').remove()
 		this.plots = []
 		copyMerge(this.settings, this.config.settings.singleCellPlot)
 
-		this.headerTr = this.table.append('tr')
-		this.tr = this.table.append('tr')
 		let sample = this.state.config.sample
+		this.legendRendered = false
+
 		if (this.state.dslabel != 'GDC') {
```
Comment (xzhou82) [inline client/plots/singleCellPlot.js:130]: can these be replaced by this?
body.sample= this.state.config.file || this.state.config.sample

Comment (airenzp) [reply client/plots/singleCellPlot.js]: No, this is a boolean to render the legend only for the first plot

Comment (airenzp) [reply client/plots/singleCellPlot.js]: Is updated now to use the variable this.tableOnPlot that it is initialized as 
`		this.tableOnPlot = appState.nav?.header_mode == 'hidden'
`

**Review by xzhou82 — APPROVED** [review]

**Pull request merged**